### ISSUE TITLE
[MNT]: make Axes.cla an alias for Axes.clear in all cases

### ISSUE THREAD

**Issue by tacaswell** [matplotlib/matplotlib#22738]
### Summary

We have clf/clear and cla/clear as pairs of related but not quite the same methods an the Figure and Axes classes.  The `clf`/`clear` pair has mostly been consolidated (and is fully in #22735 ), but the cla case is more work because we bunch of Axes subclasses in the code base.  Long term I think every Axes subclass:

 - `clear` and `cla` should be identical
 - every subclass should have both
 - we should never use cla in the docs

I think `ax.cla()` is too widely used to be deprecated solely on the aesthetics grounds so we have to carry it and if one subclass has it they all should (in interest of them being as interchangeable as possible).

Labeling this as a good first issue because a bit of grep will find all of the places in the code that need to be touched and there is (should be) no API choices in this work, but medium difficulty  because you should be comfortable moving around a new code base / have a solid grasp of OO inheritance. 

### Proposed fix

Following the pattern of #22735 

 - consolidate all implementation undef a `clear` method
 - ~add `cla = clear` in the class bodies~ add a clear method to the base class
 - ~add test that this is always true~ add a test that subclasses do not override `clf`

**Comment by neil-gurnani** [comment]
@tacaswell Myself (@neil-gurnani) and my friend @lijake8 would like to work on this issue. 
Is there any information we should be aware of / advice you would offer? 

@tacaswell

**Comment by anntzer** [comment]
Haven't thought much about it, but it may be easier to do
```python
def cla(self): return self.clear()
```
just once in the base class (and that will work correctly for all subclasses) instead of having to write `cla = clear` for each subclass (probably also add a comment to make sure this doesn't get accidentally reverted back on grounds of "simplification").

**Comment by timhoffm** [comment]
> Haven't thought much about it, but it may be easier to do

Indeed. In particular because that propagates to downstream libs, e.g. `WCSAxes` in astropy. Forcing identity in derived classes would otherwise need metaclasses, which is really not worth it.

**Comment by tacaswell** [comment]
@neil-gurnani Go for it!  This is the list of 

```
$ ack 'def cla\('
examples/misc/custom_projection.py
55:    def cla(self):

lib/matplotlib/projections/polar.py
381:    def cla(self):
679:    def cla(self):
781:    def cla(self):

lib/matplotlib/projections/geo.py
39:    def cla(self):
489:    def cla(self):

lib/matplotlib/axes/_base.py
1196:    def cla(self):

lib/matplotlib/axis.py
869:    def cla(self):

lib/matplotlib/spines.py
226:    def cla(self):

lib/matplotlib/pyplot.py
1108:def cla():

lib/mpl_toolkits/axes_grid1/axes_grid.py
40:    def cla(self):

lib/mpl_toolkits/axes_grid1/mpl_axes.py
55:    def cla(self):

lib/mpl_toolkits/axes_grid1/parasite_axes.py
20:    def cla(self):
141:    def cla(self):

lib/mpl_toolkits/axisartist/floating_axes.py
332:    def cla(self):

lib/mpl_toolkits/axisartist/axislines.py
502:    def cla(self):

lib/mpl_toolkits/mplot3d/axes3d.py
935:    def cla(self):


```

> Haven't thought much about it, but it may be easier to do

Fair, but then we can have clear and cla diverge again.


> Forcing identity in derived classes would otherwise need metaclasses, which is really not worth it.

I think we can do this with `__subclass_init__`, but agree it is probably overkill/too cute /too surprising to be worth doing.

**Comment by timhoffm** [comment]
> Fair, but then we can have clear and cla diverge again.

That's no different to the `cla = clear` case. If you really want to prohibit subclasses from defining their own `cla()`, metaclasses of `__subclass_init__` are the only choice. However, I don't think we have to enforce this. It's sufficient to delete all `cla()` from our code let the information come down from the base class. Maybe go for the slightly fancy property-way for easier testing

``` 
@property
def cla(self):
    return self.clear
```

c.f. https://github.com/matplotlib/matplotlib/pull/22735#pullrequestreview-927589969

**Comment by anntzer** [comment]
Not a huge fan of the property approach, due to https://github.com/matplotlib/matplotlib/pull/22735/files#r841278965 (and slight obfuscation).

**Comment by tacaswell** [comment]
@weiji-li Go for it!  We do not assign issues so any issue that does not have an active PR against it is fair game.

**Comment by tacaswell** [comment]
@weiji-li , @neil-gurnani , @lijake8 Can you please coordinate?

**Comment by tacaswell** [comment]
@anntzer Yeah, I found that persuasive.  I updated the top comment of this issue to follow the new method in #22735

**Comment by timhoffm** [comment]
At whoever is picking this up: The solution should be analogous to #22735.

**Comment by lijake8** [comment]
@tacaswell @timhoffm @anntzer I wanted to update you all on the status of the PR for this issue: As of earlier today, our commit is finalized and has one approving review thus far. We'd really appreciate it if you could take a look and give any feedback or a 2nd code review. Thanks!

**Comment by timhoffm** [comment]
Closed by #22802
